addon/components/bourbon-search-select-field.js

```diff
@@ -53,88 +56,39 @@ export default Component.extend(SelectMixin, {
     }
   },
 
-  focusIn() {
+  mouseDown() {
+    this.resetPrompt();
     this.set('activeOption', null);
+    this.set('showDropdown', !this.get('showDropdown'));
     this.set('inputValue', '');
   },
 
   focusOut() {
+    this.resetPrompt();
     this.set('activeOption', null);
-  },
-
-  scrollList(item, list) {
-    let listHeight = list.height();
-    let totalHeight = (this.get('activeOption') + 1) * item.scrollHeight;
-    let scrollHeight = totalHeight - listHeight;
-    list.scrollTop(scrollHeight);
-  },
-
-  keyDown(e) {
-    let el = $(e.currentTarget);
-    let list = el.find('.BourbonSelectField-menu ');
-    let allOptions = el.find(
-      '.BourbonSelectField-menu .BourbonSelectField-option'
-    );
-    let numOptions = allOptions.length;
-
-    if (e.keyCode === 40) {
-      if (this.get('activeOption') !== numOptions - 1) {
-        $(allOptions).removeClass('Bourbon--active');
-      }
-
-      if (
-        this.get('activeOption') >= 0 &&
-        this.get('activeOption') < numOptions - 1
-      ) {
-        if (this.get('activeOption') === null) {
-          this.set('activeOption', 0);
-        } else {
-          this.set('activeOption', this.get('activeOption') + 1);
-        }
-
-        this.selectOption(allOptions, list);
-      }
-    } else if (e.keyCode === 38) {
-      if (this.get('activeOption') === null) {
-        return;
-      }
-
-      if (
-        this.get('activeOption') > 0 &&
-        this.get('activeOption') < numOptions
-      ) {
-        if (this.get('activeOption') !== numOptions) {
-          $(allOptions).removeClass('Bourbon--active');
-        }
-
-        this.set('activeOption', this.get('activeOption') - 1);
-        this.selectOption(allOptions, list);
-      }
-    } else if (e.keyCode === 13) {
-      e.preventDefault();
-      this.resetOptions();
-    }
+    this.inputValueObserver();
+    this.set('showDropdown', false);
   },
 
   resetPrompt: observer('label', function() {
-    if (this.get('label')) {
+    if (this.get('value') === null && this.get('prompt')) {
+      this.set('inputValue', this.get('prompt'));
+    } else if (this.get('label')) {
       this.set('inputValue', this.get('label'));
     } else if (this.get('prompt') && this.get('inputValue') !== '') {
       this.set('inputValue', this.get('prompt'));
     }
   }),
 
-  selectOption(allOptions, list) {
-    let selectedOption = allOptions[this.get('activeOption')];
-    this.scrollList(selectedOption, list);
-    $(selectedOption).addClass('Bourbon--active');
-  },
-
-  resetOptions() {
-    this.send('updateSearchSelection');
-    this.set('showDropdown', false);
-    this.set('activeOption', null);
-    document.activeElement.blur();
+  keyDown(e) {
+    this.moveUpDown(e);
+    if (e.keyCode === 13) {
```

Comment:
could you add a comment with the corresponding key for keyCode 13?

Reply:
done